MqttBrokerConnectionTests unstable

See e.g. Travis https://travis-ci.org/openhab/openhab-core/jobs/598834457

```
[ERROR] timeoutWhenNotReachableFuture(org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests)  Time elapsed: 0.205 s  <<< FAILURE!
org.mockito.exceptions.verification.NeverWantedButInvoked: 
connectionCallback.onFailure(<any>, <any>);
Never wanted here:
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.timeoutWhenNotReachableFuture(MqttBrokerConnectionTests.java:225)
But invoked here:
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection.lambda$1(MqttBrokerConnection.java:733)
	at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.timeoutWhenNotReachableFuture(MqttBrokerConnectionTests.java:225)
```

One more (see Travis https://travis-ci.org/openhab/openhab-core/jobs/600039891).

```
[ERROR] reconnectPolicy(org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests)  Time elapsed: 0.067 s  <<< FAILURE!
org.mockito.exceptions.verification.WantedButNotInvoked: 

Wanted but not invoked:

mqttBrokerConnectionEx.start();
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:167)

However, there were exactly 7 interactions with this mock:
mqttBrokerConnectionEx.setConnectionCallback(
    org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionEx$MockitoMock$1253350811@4aa3d36
);
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:143)

mqttBrokerConnectionEx.setReconnectStrategy(
    org.eclipse.smarthome.io.transport.mqtt.reconnect.PeriodicReconnectStrategy@2d140a7
);
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:146)

mqttBrokerConnectionEx.getReconnectStrategy();
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:147)

mqttBrokerConnectionEx.setReconnectStrategy(
    org.eclipse.smarthome.io.transport.mqtt.reconnect.PeriodicReconnectStrategy$MockitoMock$1533906995@2aa27288
);
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:156)

mqttBrokerConnectionEx.cancelTimeoutFuture();
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection$ConnectionCallback.onFailure(MqttBrokerConnection.java:148)

mqttBrokerConnectionEx.connectionState();
-> at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection$ConnectionCallback.onFailure(MqttBrokerConnection.java:152)

mqttBrokerConnectionEx.getHost();
-> at org.eclipse.smarthome.io.transport.mqtt.reconnect.PeriodicReconnectStrategy.lambda$0(PeriodicReconnectStrategy.java:116)

	at org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnectionTests.reconnectPolicy(MqttBrokerConnectionTests.java:167)
```

--- a/bundles/org.openhab.core.io.transport.mqtt/pom.xml
+++ b/bundles/org.openhab.core.io.transport.mqtt/pom.xml
@@ -18,6 +18,12 @@
       <artifactId>org.openhab.core.config.core</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.openhab.core.bundles</groupId>
+      <artifactId>org.openhab.core.test</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 
 </project>

--- a/bundles/org.openhab.core.io.transport.mqtt/src/test/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnectionTests.java
+++ b/bundles/org.openhab.core.io.transport.mqtt/src/test/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnectionTests.java
@@ -31,6 +31,7 @@ import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.io.transport.mqtt.internal.client.MqttAsyncClientWrapper;
 import org.eclipse.smarthome.io.transport.mqtt.reconnect.AbstractReconnectStrategy;
 import org.eclipse.smarthome.io.transport.mqtt.reconnect.PeriodicReconnectStrategy;
+import org.eclipse.smarthome.test.java.JavaTest;
 import org.junit.Test;
 import org.osgi.service.cm.ConfigurationException;
 
@@ -43,7 +44,7 @@ import com.hivemq.client.mqtt.mqtt3.message.publish.Mqtt3Publish;
  * @author David Graeff - Initial contribution
  * @author Jan N. Klug - adjusted to HiveMQ client
  */
-public class MqttBrokerConnectionTests {
+public class MqttBrokerConnectionTests extends JavaTest {
     @Test
     public void subscribeBeforeOnlineThenConnect()
             throws ConfigurationException, MqttException, InterruptedException, ExecutionException, TimeoutException {
@@ -168,8 +169,7 @@ public class MqttBrokerConnectionTests {
 
         // Check lostConnect
         verify(mockPolicy).lostConnection();
-        Thread.sleep(10);
-        verify(connection).start();
+        waitForAssert(() -> verify(connection).start());
         assertTrue(mockPolicy.isReconnecting());
 
         // Fake connection established

